chore(AWSLex): Dropped support for arm64 iphonesimulator (#3365)
aws-amplify/aws-sdk-ios: CHANGELOG.md

-Features for next release

### Misc. Updates
 - AWSMobileClient

>     - Created a new target named `AWSMobileClientXCF` for the support of XCFramework. [PR #3334](https://github.com/aws-amplify/aws-sdk-ios/pull/3334)

Reviewer: ```suggestion
    - Created a new target named `AWSMobileClientXCF` to support building AWSMobileClient as an XCFramework. [PR #3334](https://github.com/aws-amplify/aws-sdk-ios/pull/3334)
```

 - AWSLex

>     - Updated the AWSLex with the latest version of libBlueAudioSourceiOS and added arm64 iphonesimulator into Excluded Architecture. [PR #3365](https://github.com/aws-amplify/aws-sdk-ios/pull/3365)

Reviewer: Note this as a breaking change according to our normal breaking change format.

Author: As discussed, this is not a breaking change. We are just not enabling arm64 support for AWSLex.
